openmrs/omod/src/main/java/org/openmrs/projectbuendia/webservices/rest/RequestUtil.java

```diff
@@ -0,0 +1,40 @@
+// Copyright 2015 The Project Buendia Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License.  You may obtain a copy
+// of the License at: http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distrib-
+// uted under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+// OR CONDITIONS OF ANY KIND, either express or implied.  See the License for
+// specific language governing permissions and limitations under the License.
+
+package org.openmrs.projectbuendia.webservices.rest;
+
+import org.openmrs.module.webservices.rest.web.RequestContext;
+import org.openmrs.projectbuendia.Utils;
+
+import javax.annotation.Nullable;
+import java.text.ParseException;
+import java.util.Date;
+
+/**
+ * Utilities for working with requests and request parameters.
+ */
+public class RequestUtil {
+    /**
+     * Obtains the "Sync From" date from a request. This is stored in the "since" HTTP parameter
+     * in ISO 8601 format.
+     * @param context the relevant request.
+     * @return the instant that data should be provided from (inclusive), or {@code null} if the
+     * "since" parameter was not present in the request.
+     */
+    @Nullable
+    public static Date getSyncFromDate(RequestContext context) throws ParseException {
+        String param = context.getParameter("since");
+        if (param == null) {
+            return null;
+        }
+        return Utils.fromIso8601(param);
```

Comment:
The parse/format time is something that I hadn't thought about. I suspect that the time it takes to do that is dwarfed by the rest of the request, though (JSON, I'm looking at you!).

I like the use of ISO-8601 because it makes debugging the REST API heaps easier, especially for incremental sync, where correct timestamps are critical. Initially, I'd designed the protocol to use millis since epoch in UTC, but I made the following programming errors:
- At one point, I was supplying epoch time in seconds, instead of millis, and so more results than I was expecting were being returned. This would not be a problem using a human-readable format.
- I was never really 100% sure what timezone I was operating in. The fact that ISO-8601 makes this explicit is awesome.

I think I'd like to retain ISO-8601 for the API for ease of debugging at minimal performance cost, but to make the protocol explicit in saying "only the UTC timezone is accepted".

Does that sound reasonable to you?

Also, FWIW, it seems like readable / standards-based date/time formats are much more common for REST APIs than millis (e.g. Facebook uses ISO 8601, Twitter uses RFC5322)